tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFile.java

```diff
@@ -1462,22 +1463,70 @@ public static Set<String> getMatchingXMLFiles(File sourceDirs[], Matcher m) {
         return s;
     }
 
-    @Override
-    public Iterator<String> iterator() {
+    private final boolean DEFAULT_ITERATION_INCLUDES_EXTRAS = true;
+
+    public Iterator<String> iteratorDefault() {
+        if (DEFAULT_ITERATION_INCLUDES_EXTRAS) {
+            return Iterators.filter(fullIterable().iterator(), p -> getStringValue(p) != null);
```

Comment:
fullIterable is unchanged. If DEFAULT_ITERATION_INCLUDES_EXTRAS is true, iteratorDefault is the same as fullIterable except that it skips paths with null values